usr/share/rear/lib/opaladmin-workflow.sh

```diff
@@ -373,6 +409,30 @@ function opaladmin_device_unlock_if_locked() {
     fi
 }
 
+function opaladmin_deactivate_locking() {
+    local device="${1:-?}"
+    local identification="${2:-?}"
+    # permanently deactivates the locking mechanism on the device.
+
+    LogUserOutput "Persistently deactivating the locking mechanism on device $identification..."
+    opaladmin_get_disk_password
+    opal_device_deactivate_locking "$device" "$OPAL_DISK_PASSWORD"
+    StopIfError "Could not deactivate locking on device \"$device\"."
```

Comment:
Only FYI
in general about possible problems when using the old `...IfError` functions see
https://github.com/rear/rear/pull/2443#discussion_r448166732
in particular therein also follow the links to the other issues

Reply:
If I understand the situation correctly, current code in TCG Opal-related scripts is not affected by `StopIfError` problems. However, it could be affected in the future once someone decides to insert a seemingly innocent change (such as referring to `?$`). So if you'd like to, I could change invocations to something like
```
opal_device_deactivate_locking "$device" "$OPAL_DISK_PASSWORD" ||
Error "Could not deactivate locking on device \"$device\"."
```

That way there would be no problems with line length or extra statements which would otherwise clutter up the primary code.

If you'd prefer this, don't merge now and I'll push another commit first.

Reply:
@OliverO2 
I also think that the current code is not affected by `...IfError` problems.

Therefore I mentioned it oly as "FYI" not as a request to change it.

If you like you could change it and I wait with merging it
or
I merge it as is and later you could do a separated pull request
where you could clean up all `...IfError` usage in current OPAL code:
```
# find usr/sbin/rear usr/share/rear/ -type f | xargs grep 'IfError ' | grep -i '/opal'

usr/share/rear/build/OPALPBA/Linux-i386/105_reinclude_essential_files.sh:
StopIfError "Could not re-include essential files"

usr/share/rear/lib/opal-functions.sh:
    StopIfError "Could not set up $device."
    StopIfError "Could not reset data encryption key (DEK) of Opal 2 disk $device."
    StopIfError "Could not enable the shadow MBR on Opal 2 disk $device."
    StopIfError "Could not upload the PBA image \"$pba_image_file\" to Opal 2 disk $device."

usr/share/rear/lib/opaladmin-workflow.sh:
                StopIfError "Could not set up device \"$device\"."
                    StopIfError "Could not enable the shadow MBR on device \"$device\"."
                    StopIfError "Could not upload the PBA image to device \"$device\"."
                StopIfError "Could not upload the PBA image to device \"$device\"."
                StopIfError "Could not reset device \"$device\" to factory defaults."
        StopIfError "Could not unlock device \"$device\"."
```

Interesting syntax example that `||` continues on next line
where additional indentation should be used to make things more clear like
```
opal_device_deactivate_locking "$device" "$OPAL_DISK_PASSWORD" ||
    Error "Could not deactivate locking on device '$device'."
```
Here I also used single quotes within double quotes
because one does not need to escape single quotes within double quotes
which makes the code easier to read (at least for me).
By the way:
When I show a value in a message I use single quotes
when the value could be empty like
```
Log "Thingy '$thingy' does not exist"
```
so that one can see in the message when the value is empty like
```
Thingy '' does not exist
```
or when the value has unexpected spaces like
```
File ` somefile` does not exist
```
and proper quoting is used in the code like
```
# somefile=' /etc/fstab'
# ls "$somefile"
ls: cannot access ' /etc/fstab': No such file or directory
```
;-)

Reply:
@jsmeix 
Let's do this within this PR. Saves some bureaucracy, leaving more time to code and test.

I'll look at those quotes, too.

Reply:
Fine! No rush, take your time.

Reply:
With fe0be18f42d25525d5fd582a364eb3aa8161bed9 things should look a bit nicer now.